power_grid_model_c/power_grid_model/include/power_grid_model/optimizer/tap_position_optimizer.hpp

```diff
@@ -23,6 +23,7 @@ using TrafoGraphIdx = Idx;
 using EdgeWeight = int64_t;
 using WeightedTrafo = std::pair<Idx2D, EdgeWeight>;
 using WeightedTrafoList = std::vector<WeightedTrafo>;
+using RankedTransformerGroups = std::vector<std::vector<Idx2D>>;
```

Comment:
the more i think about it, the more i'm starting to think that this might not be the correct approach after all. we shouldn't need the trafo's `Idx2D` but the regulator's `Idx` (which is equivalent to `Idx2D{.group = Container::template get_type_idx<AutomaticTapRegulator>, .idx = regulator_idx}` because of how the container works)

in addition, i'm also thinking more and more about how we actually want to know not only the rank but also which connected component the regulator belongs to...

I am quite hesitant to make a vector of vectors of vectors but i am not sure if we can come up with a better way. any thoughts?

Reply:
The other option I had in mind is to simply have an ordered list, with additional edge weight / rank associated with each entry. That makes basically everything else more complicated.

`{ [id=*, rank=100], ...,[id=*, rank=10], [id=*, rank=10], [id=*, rank=10], ...}`

Reply:
that's OK too. i FINALLY found a way to handle different transformer types nicely in my branch. will push now so you can have a look

Reply:
pushed to https://github.com/PowerGridModel/power-grid-model/pull/557

Reply:
out of scope of this PR